app/src/main/java/de/grobox/transportr/trips/search/DirectionsActivity.java

```diff
@@ -126,6 +128,30 @@ private void processIntent() {
 				case TASK_WORK:
 					to = viewModel.getWork().getValue();
 					break;
+				case TASK_FAVORITE:
+					long locationId, fromId, toId;
+					Long viaId;
+					fromId = intent.getLongExtra(FROM, 0);
+					viaId = (Long) intent.getSerializableExtra(VIA);
+					toId = intent.getLongExtra(TO, 0);
+
+					from = null;
+					to = null;
+
+					try{
+						for (FavoriteLocation location : viewModel.getLocations().getValue()) {
+							locationId = location.getUid();
```

Comment:
You need to ask the view model for the FavoriteTripItem with the `uid` from the intent. If the data is not yet available when the intent is processed, you need to observe it to get notified when available.